components/membership/membership-rest/src/main/kotlin/net/corda/membership/rest/v1/types/request/HostedIdentitySetupRequest.kt

```diff
@@ -6,12 +6,23 @@ package net.corda.membership.rest.v1.types.request
  * @param p2pTlsCertificateChainAlias The certificates chain alias.
  * @param useClusterLevelTlsCertificateAndKey Should the cluster-level P2P TLS certificate type and key be used or
  *     the virtual node certificate and key.
- * @param sessionKeyId The session key identifier (will use the first one if null).
- * @param sessionCertificateChainAlias The certificate chain alias of the Session Key. Should be null if no PKI is used for sessions.
+ * @param sessionKeys The list of session keys and certificates. If empty the first session key will be used.
  */
 data class HostedIdentitySetupRequest(
     val p2pTlsCertificateChainAlias: String,
     val useClusterLevelTlsCertificateAndKey: Boolean?,
-    val sessionKeyId: String?,
-    val sessionCertificateChainAlias: String? = null,
+    val sessionKeys: List<HostedIdentitySessionKey> = emptyList(),
```

Comment:
```suggestion
    val sessionKeysAndCerts: List<HostedIdentitySessionKeyAndCert>,
```
I can't think of a scenario where a user would want to specify no keys at all, so empty list doesn't look like a great default if that's the case.

Reply:
We should also validate that btw, now that we have a list as I don't think it can be validated as part of the JSON schema.

Reply:
The validation seems to work.

Reply:
which validation works? I referred to verifying the list contains at least one element.

Reply:
We have this validation in the rest implementation.

Reply:
at which place exactly? Looking at `NetworkRestResourceImpl.setupHostedIdentities()` I can see we are accepting empty list from the user without returning an error.

The conversation [here](https://github.com/corda/corda-runtime-os/pull/3394#discussion_r1140329312) is about the same thing, I believe. I am not sure I got my point across there. I think we shouldn't be allowing the operator to specify empty lists (even when there's a single session key in the cluster). The REST endpoint should mandate at least one session initiation key be specified, given the sensitivity of these operations.